internal/cli/cli.go

```diff
@@ -54,66 +57,66 @@ func main() {
 		}
 
 		// Data Field Header
-		dataHeader := TMDataFieldHeader{}
+		dataHeader := pus.TMDataFieldHeader{}
 		dhsize := binary.Size(dataHeader)
 		dhr := bytes.NewReader(buf[0:])
-		err = dataHeader.read(dhr)
+		err = dataHeader.Read(dhr)
 		if err != nil {
 			log.Fatal(err)
 		}
 
 		// Data field  header
-		fmt.Printf("%04b ", dataHeader.getPUS())
+		fmt.Printf("%04b ", dataHeader.PUSVersion())
 
-		fmt.Printf("%4d ", dataHeader.getServiceType())
-		fmt.Printf("%4d ", dataHeader.getServiceSubType())
+		fmt.Printf("%4d ", dataHeader.ServiceType)
+		fmt.Printf("%4d ", dataHeader.ServiceSubType)
 		if messageType == 0 {
-			fmt.Printf("%d ", dataHeader.getTime())
+			fmt.Printf("%d ", dataHeader.Time())
 		}
 
 		// if crc16.ChecksumCCITTFalse(append(header, data[:dataPackageLength-2]...)) == binary.BigEndian.Uint16(data[dataPackageLength-2:dataPackageLength]) {
 		// 	fmt.Print("✔️ ")
 		// } else {
 		// 	fmt.Print("❌ ")
 		// }
-		if dataHeader.getServiceType() == 3 && dataHeader.getServiceSubType() == 25 {
+		if dataHeader.ServiceType == 3 && dataHeader.ServiceSubType == 25 {
 			sid := binary.BigEndian.Uint16(buf[dhsize:])
 			r := bytes.NewReader(buf[dhsize+2:])
 			if sid == 1 {
-				stat := STAT{}
-				err = stat.read(r)
+				stat := pus.STAT{}
+				err = stat.Read(r)
 				if err != nil {
 					log.Fatal("stat", err)
 				}
 				fmt.Print(" STAT: ", stat)
 			}
 			if sid == 10 {
-				htr := HTR{}
-				err = htr.read(r)
+				htr := pus.HTR{}
```

Comment:
This should be possible to refactor since the structure is identical to sid == 1, it should be possible to have a mapping of sid to parser.

Reply:
yes